test/calibration/experiments/test_drag.py

```diff
@@ -96,6 +97,27 @@ def test_end_to_end(self):
         self.assertTrue(abs(result.value.n - backend.ideal_beta) < self.test_tol)
         self.assertEqual(result.quality, "good")
 
+    @data(
+        (0.0050, 1.0, 0.00, [1, 3, 5], None),  # partial oscillation.
+        (0.0025, 0.5, 0.00, [1, 3, 5], None),  # even slower oscillation with amp < 1
+        (0.0050, 0.8, 0.05, [3, 5, 7], None),  # constant offset, i.e. lower SNR.
+        (0.1000, 0.5, 0.10, [1, 3, 5], np.linspace(-1, 1, 51)),  # Beta not in range
+    )
+    @unpack
+    def test_nasty_data(self, error, amp, offset, reps, betas):
+        """A set of tests for non-ideal data."""
+        backend = DragBackend(error=error, gate_name="Drag(xp)", max_prob=amp, offset_prob=offset)
+
+        drag = RoughDrag(0, self.x_plus, betas=betas)
+        drag.set_experiment_options(reps=reps)
+
+        exp_data = drag.run(backend)
+        self.assertExperimentDone(exp_data)
+        result = exp_data.analysis_results(1)
```

Comment:
Has curve analysis always produced two analysis results?

```suggestion
        result = exp_data.analysis_results("beta")
```